inclusive-design/covid-assessment-centres
# feat: auto download and commit new data files published by ODS (resolves #5)

## Description

A temporary solution that uses Github actions to schedule a cron job that automatically downloads and commits new data files published by ODS into this repository.

According to the Github policy, if a repository stays idle for 60 days, cron jobs will be stopped. This limitation determines using Github cron job is only a temporary solution. A long term solution targeting at hosting our own cron job independently from Github is in development.

This PR is based on https://github.com/inclusive-design/covid-assessment-centres/pull/9, which should be merged first.

It would be easier to review this PR by looking at [the diff](https://github.com/cindyli/covid-assessment-centres/compare/add-linting-tool...cindyli:auto-update-script) that excludes the change in https://github.com/inclusive-design/covid-assessment-centres/pull/9.

## Steps to test

Note that scheduled Github workflows only run on the default or base branch. See [the Github doc](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events). To test this PR, consider to push it into reviewer's personal main branch.

1. The cron job should run once a day;
2. When no new file is detected, the cron job completes successfully;
3. When a new file is detected, the cron job downloads and commits the new data file into the `/ODC` directory. `/ODC/latest.json` is updated accordingly with the name of the new file. The commit is using `wecountproject` account.

Step 3 can be tested by:
1. Remove `ODC/assessment_centre_locations_2020_08_20.csv`
2. Modify the content of `./ODC/latest.json` to point to a different filename;
3. Push changes above into the personal main branch.
